core/src/block_data_manager.rs

```diff
@@ -63,6 +64,47 @@ impl HeapSizeOf for EpochExecutionContext {
     fn heap_size_of_children(&self) -> usize { std::mem::size_of::<u64>() }
 }
 
+#[derive(Clone)]
+pub struct ConsensusGraphExecutionInfo {
+    pub state_valid: bool,
+    pub original_deferred_state_root: H256,
+    pub original_deferred_receipt_root: H256,
+    pub original_deferred_logs_bloom_hash: H256,
+}
+
+impl Default for ConsensusGraphExecutionInfo {
+    fn default() -> Self {
+        ConsensusGraphExecutionInfo {
+            state_valid: true,
+            original_deferred_state_root: Default::default(),
+            original_deferred_receipt_root: Default::default(),
+            original_deferred_logs_bloom_hash: Default::default(),
+        }
+    }
+}
+
+impl Encodable for ConsensusGraphExecutionInfo {
```

Comment:
Could use rlp derive way to simplify the code, please refer to sync message "GetBlockHeaders"